diff/walking/differ.go

```diff
@@ -118,9 +118,16 @@ func (s *walkingDiff) Compare(ctx context.Context, lower, upper []mount.Mount, o
 			if isCompressed {
 				dgstr := digest.SHA256.Digester()
 				var compressed io.WriteCloser
-				compressed, errOpen = compression.CompressStream(cw, compression.Gzip)
-				if errOpen != nil {
-					return errors.Wrap(errOpen, "failed to get compressed stream")
+				if config.Compressor != nil {
```

Comment:
I guess this should just ignore the mediatype and not check on `isCompressed`.

```
if config.Compressor != nil {
} else {
  if isCompressed {
  }
}
```

Reply:
Thank you for the review. Fixed to always perform compression if `config.Compressor != nil` (addressed in the above `switch` statement instead of here). LGTY?